python install of master fails on fedora since merge of 4501

since https://github.com/OSGeo/gdal/pull/4501 was merged, `make install` fails in swig/python for me with this:

```
...
make[1]: Entering directory '/home/vincent/src/gdal/gdal/swig'
for dir in python ; do (cd $dir; make install) || exit; done
make[2]: Entering directory '/home/vincent/src/gdal/gdal/swig/python'
----------------------------------------------------------------------

The GDAL python package has been installed in lib/python3/dist-packages
Please ensure to add lib/python3/dist-packages to your PYTHONPATH

----------------------------------------------------------------------
env PYTHONPATH=lib/python3/dist-packages${PYTHONPATH:+:$PYTHONPATH} \
       python setup.py install --single-version-externally-managed --record=record.txt --prefix=/usr/local --install-layout=deb
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
  or: setup.py --help [cmd1 cmd2 ...]
  or: setup.py --help-commands
  or: setup.py cmd --help

error: option --install-layout not recognized
make[2]: *** [GNUmakefile:142: install] Error 1
```

Probably because now the script uses a different site_package_dir (due to the change in GNUMakefile:105), which does trigger the check in line 135, and the debian-specific "--install-layout" option is added. Which fails on fedora, obviously.

--- a/gdal/swig/python/trimmedsysconfig.py
+++ b/gdal/swig/python/trimmedsysconfig.py
@@ -23,10 +23,8 @@ BASE_EXEC_PREFIX = os.path.normpath(sys.base_exec_prefix)
 
 # Added by GDAL
 def _is_debian():
-    import setuptools.command.easy_install
-    ei = setuptools.command.easy_install.easy_install
-    for opt, _, _ in ei.user_options:
-        if opt == 'install-layout=':
+    for p in sys.path:
+        if 'dist-packages' in p:
             return True
     return False
 

